packages/core/src/resource/resource.ts

```
@@ -162,7 +167,20 @@ export class ResourceImpl<T, R> extends BaseWritableResource<T> implements Resou
       computed(
         () => {
           const streamValue = this.state().stream?.();
-          return streamValue && isResolved(streamValue) ? streamValue.value : this.defaultValue;
+
+          if (!streamValue) {
+            return defaultValue;
+          }
+
+          if (!isResolved(streamValue)) {
+            if (defaultValue !== undefined) {
```

It should return the default value if provided when there is an error. I got this logic from https://github.com/angular/angular/blob/57bde58b9e5b5ad1bd26fef93429f4376b0349b9/packages/core/src/resource/api.ts#L188-L192

---

I think it would be more consistent that resource in Error state should always throw on value read.  
That would also mean, rephrasing that comment.

---

@alxhub I don't think this aligns with our new thinking that `.value()` should throw if it's in the error state.

---

Changed a comment and applied a logic so that when there is an error it would throw instead of returning a deafault value.